src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -761,9 +767,13 @@ static int parse_request(uint8_t code, size_t len, const void *option, void *use
 
                 break;
         case SD_DHCP_OPTION_RELAY_AGENT_INFORMATION:
-                req->agent_info_option = (uint8_t*)option - 2;
-
+                if (req->agent_info_option)
+                        mfree(req->agent_info_option);
+                req->agent_info_option = new(uint8_t, len);
+                memcpy(req->agent_info_option, option, len);
```

Comment:
memdup()?

Reply:
Didn't know about this one, thanks. Done.